test/unit/test_handler_mocking_openidc.lua

```diff
@@ -405,6 +406,67 @@ function TestHandler:test_authenticate_with_session_cookie_samesite_set_to_none(
   lu.assertItemsEquals(v, opts.session)
 end
 
+function TestHandler:test_authenticate_ok_to_force_authentication_path()
+  -- arrange
+  local actual_unauth_action
+  ngx.var.request_uri = idpAuthPath
+
+  -- mock authenticate to be able to check unauth_action
+  self.module_resty.openidc.authenticate = function(opts, target_url, unauth_action)
+    actual_unauth_action = unauth_action
+    return {}, false, "/", session
+  end
+  -- act
+  self.handler:access({ force_authentication_path = idpAuthPath })
+
+  -- assert
+  lu.assertTrue(self:log_contains("login request detected"))
+  lu.assertEquals(actual_unauth_action, nil)
+end
+
+function TestHandler:test_authenticate_ok_to_non_force_authentication_path()
+  -- arrange
+  local actual_unauth_action
+
+  -- mock authenticate to be able to check unauth_action
+  self.module_resty.openidc.authenticate = function(opts, target_url, unauth_action)
+    actual_unauth_action = unauth_action
+    return {}, false, "/", session
+  end
+  -- act
+  self.handler:access({ force_authentication_path = idpAuthPath })
+
+  -- assert
+  lu.assertEquals(actual_unauth_action, "pass")
+end
+
+function TestHandler:test_authenticate_ok_to_force_authentication_path()
```

Comment:
this function has the same name - is that ok?

Reply:
addressed